vis_methods.py

```diff
@@ -297,3 +297,165 @@ def plot(self, alternate_plot):
             ax.set_xticklabels(x_labels, rotation=30)
             ax.set_title(name)
             plt.show()
+
+
+class ManyBenchmarks(Visualiser):
+    """
+    A visualiser state that plots results for two different benchmarks
+    (for example time and accuracy) of the same set of metrics against
+    repository commits for all supplied metrics.
+
+    """
+    def __init__(self, vis):
+        """
+        A visualiser that provides functionality to visualise two benchmarks of
+        tehuti metric results over a series of commits to an underlying
+        GitHub repo.
+
+        Arg:
+
+        * vis:
+            The :class:`tehuti-vis.Vis` class that this class is providing
+            a visualiser state for.
+
+        """
+        self.vis = vis
+
+    def select_data(self, commits, metrics):
+        """
+        The select data method for this state.
+
+        Select and re-format data from the supplied metrics results file.
+
+        Data is selected based on all specified `commits` and `metrics`. Data
+        is formatted into the standard format used by plot states, with
+        specific formatting that converts any single values into one-element
+        lists.
+        If no commits or metrics are specified then all commits or metrics
+        in the supplied metrics results file are selected.
+
+        Args:
+
+        * commit:
+            One or more valid repository commits that have been benchmarked.
+        * metrics:
+            One or more metrics that have results in the metrics results file.
+
+        Returns:
+            The selected data formatted for plotting.
+
+        """
+        commits, metrics = self._select_data_common(commits, metrics)
+        benchmarks = {}.fromkeys([m.split('-')[0] for m in metrics]).keys()
+        metrics = {}.fromkeys([m.split('-', 1)[1] for m in metrics]).keys()
```

Comment:
Aha, what's missing is the comment that describes why I've used that bizarre Python idiom. What it's doing is getting only the unique values within the list:
```python
>>> l = ['a-5', 'b-6', 'c-3', 'a-5', 'd-1', 'c-3']
>>> print [i.split('-')[0] for i in l]
['a', 'b', 'c', 'a', 'd', 'c']
>>> print {}.fromkeys([i.split('-')[0] for i in l]).keys()
['a', 'c', 'b', 'd']
```
Useful, huh?